src/NUnitFramework/tests/Attributes/TestOrderAttributeTests.cs

```diff
@@ -22,6 +24,51 @@ public void CheckOrderIsCorrect()
             Assert.AreEqual(work.Children[1].Test.Name, "Y_SecondTest");
             Assert.AreEqual(work.Children[2].Test.Name, "Z_ThirdTest");
         }
-        
+
+        [Test]
+        [TestCaseSource(nameof(Cases))]
+        public void CheckClassOrderIsCorrect(List<Type> candidateTypes)
+        {
+            var testSuite = TestBuilder.MakeFixture(candidateTypes);
+
+            var work = TestBuilder.PrepareWorkItem(testSuite, null) 
+                as CompositeWorkItem;
+
+            var fixtureWorkItems = 
+                ((work.Children[0] as CompositeWorkItem)
+                .Children[0] as CompositeWorkItem)
+                .Children;
+
+            Assert.AreEqual(candidateTypes.Count, fixtureWorkItems.Count);
+            for (var i = 1; i < fixtureWorkItems.Count; i++)
+            {
+                var previousTestOrder = int.Parse(fixtureWorkItems[i - 1]
+                    .Test.Properties.Get(PropertyNames.Order).ToString());
+                var currentTestOrder = int.Parse(fixtureWorkItems[i]
+                    .Test.Properties.Get(PropertyNames.Order).ToString());
+
+                Assert.IsTrue(previousTestOrder < currentTestOrder);
+            }
+        }
+
+        private static readonly object[] Cases =
+        {
```

Comment:
Will the tests still pass if we supply the types in a different order? As the code is now, the classes are already sorted in the source code and here we also supply the classes in a sorted manner (or am I missing something?)